loader/src/loader/updater.cpp

```diff
@@ -82,33 +82,37 @@ void updater::fetchLatestGithubRelease(
 }
 
 void updater::downloadLatestLoaderResources() {
-    log::debug("Downloading latest resources", Loader::get()->getVersion().toString());
+    // Log the start of the download process
+    log::debug("Downloading latest resources for version: ", Loader::get()->getVersion().toString());
+
+    // Fetch the latest GitHub release
     fetchLatestGithubRelease(
+        // Lambda function to process the JSON response
         [](matjson::Value const& raw) {
             auto json = raw;
             JsonChecker checker(json);
             auto root = checker.root("[]").obj();
 
-            // find release asset
+            // Find the release asset named "resources.zip"
             for (auto asset : root.needs("assets").iterate()) {
                 auto obj = asset.obj();
-                if (obj.needs("name").template get<std::string>() == "resources.zip") {
-                    updater::tryDownloadLoaderResources(
-                        obj.needs("browser_download_url").template get<std::string>(),
-                        false
-                    );
+                std::string assetName = obj.needs("name").template get<std::string>();
+                if (assetName == "resources.zip") {
```

Comment:
were these changes really necessary? no point in storing the string in a local variable, and also this might be implicitly creating a copy